src/checkers/inference/solver/frontend/Lattice.java

```diff
@@ -0,0 +1,96 @@
+package checkers.inference.solver.frontend;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * Lattice class contains necessary information about qualifier hierarchy for
+ * constraint constraint solving.
+ * 
+ * @author jianchu
+ *
+ */
+public class Lattice {
```

Comment:
Hi @Jianchu , since `Lattice` is a general class that frequently and commonly used in the Solver framework, can you write some javadocs for this class? Specifically, for each field, a javadoc describes what this field is would be very helpful.